packages/astro/src/core/create-vite.ts

```diff
@@ -112,7 +115,13 @@ export async function createVite(
 		customLogger: createViteLogger(logger, settings.config.vite.logLevel),
 		appType: 'custom',
 		optimizeDeps: {
-			entries: ['src/**/*'],
+			// Scan all files within `srcDir` except for known server-code (e.g endpoints)
+			entries: [
+				`${srcDirPattern}!(pages)/**/*`, // All files except for pages
```

Comment:
I made some tests and it seems that `srcDirPattern` will always have a trailing slash. The only case I'm not sure is on Windows.